controllers/machineset_controller.go

```diff
@@ -386,7 +386,7 @@ func shouldExcludeMachine(machineSet *clusterv1.MachineSet, machine *clusterv1.M
 
 // adoptOrphan sets the MachineSet as a controller OwnerReference to the Machine.
 func (r *MachineSetReconciler) adoptOrphan(machineSet *clusterv1.MachineSet, machine *clusterv1.Machine) error {
-	patch := client.MergeFrom(machine)
+	patch := client.MergeFrom(machine.DeepCopy())
```

Comment:
:grimacing: Since this goes back to my patch updates I went ahead to see if I messed up any other places, and it turns out I did: https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machinedeployment_sync.go#L106

Reply:
Great catch! Added to the PR